### ISSUE TITLE
`no-unregistered-classes`: `ignore` option matches substrings; should require exact class match

### ISSUE THREAD

**Issue by yuheiy** [schoero/eslint-plugin-better-tailwindcss#218]
`no-unregistered-classes` rule’s `ignore` option appears to filter by **substring** rather than an **exact class token**.
As a result, values that only *contain* the ignored string are also excluded.

When `ignore: ['prose']` is set, a class like `prosea` is also ignored, even though only `prose` should be excluded.

`eslint.config.js`:

```js
import betterTailwindcss from 'eslint-plugin-better-tailwindcss';

export default [
  {
    languageOptions: {
      parserOptions: { ecmaFeatures: { jsx: true } },
    },
    plugins: {
      'better-tailwindcss': betterTailwindcss,
    },
    rules: {
      'better-tailwindcss/no-unregistered-classes': [
        'warn',
        {
          ignore: ['prose'], // expect to ignore ONLY the exact class "prose"
        },
      ],
    },
    settings: {
      'better-tailwindcss': {
        entryPoint: 'index.css',
      },
    },
  },
];
```

`test.js`:

```jsx
<div className="prosea"></div>;
<div className="flexa"></div>;
```

Actual result:

```console
yuhei eslint-plugin-better-tailwindcss-test % npx eslint .

/Users/yuhei.yasuda/Desktop/eslint-plugin-better-tailwindcss-test/test.js
  2:17  warning  Unregistered class detected: flexa  better-tailwindcss/no-unregistered-classes

✖ 1 problem (0 errors, 1 warning)
```

`prosea` (line 1) is **not reported** — it gets ignored because it contains `prose` as a substring.

**Comment by yuheiy** [comment]
Thanks for clarifying. `ignore: ["prose$"]` fits my use case, so I’ll use it.
